hw/ip/hmac/lint/hmac.waiver

```diff
@@ -18,3 +18,8 @@ waive -rules {NOT_READ} -location {*_reg_top.sv} -regexp {(address|param|user)}
 
 waive -rules {NOT_READ HIER_NET_NOT_READ} -location {hmac.sv} -regexp {'msg_fifo_wmask.* is not read} \
       -comment "Only first bit in a byte is used"
+
+waive -rules {CONST_FF} -location {prim_sha2.sv} -regexp {Flip-flop 'round_q\[6\]' is driven by constant zero} \
+      -comment "prim_sha2 is not instantiated in SHA2-384/512 multimode"
+waive -rules {CONST_FF} -location {prim_sha2_pad.sv} -regexp {Flip-flop 'digest_mode_flag' is driven by constant zeros in module 'prim_sha2_pad' \(MultimodeEn=1'h0\)} \
+      -comment "prim_sha2 is not instantiated in SHA2-384/512 multimode"
```

Comment:
I agree with the waiver but think the comment is unintuitive; "prim_sha2 is instantiated without SHA2-384/512 multimode (MultimodeEn=0)" or similar would be clearer IMO. But I don't think this is critical enough to block this PR

Reply:
Thanks, I had ignored these 2 since they were labelled warnings and just stuck to fixing and waiving the errors :)

Reply:
As discussed with @gdessouky, she'll make an RTL fix so we can get rid of these two rules

Reply:
Putting up a PR now with minor RTL fixes so we don't need the waivers for these warnings.